rom/dev/src/flow/cold_reset/mod.rs

```diff
@@ -73,16 +73,31 @@ impl ColdResetFlow {
     }
 }
 
-pub fn copy_tbs(tbs: &[u8], tbs_type: TbsType) -> CaliptraResult<()> {
+/// Copies the TBS to DCCM
+///
+/// # Arguments
+/// * `tbs` - TBS to copy
+/// * `tbs_type` - Type of TBS
+/// * `env` - ROM Environment
+///
+/// # Returns
+///     CaliptraResult
+pub fn copy_tbs(tbs: &[u8], tbs_type: TbsType, env: &mut RomEnv) -> CaliptraResult<()> {
     let dst = match tbs_type {
-        TbsType::LdevidTbs => unsafe {
-            let ptr = &mut LDEVID_TBS_ORG as *mut u8;
-            core::slice::from_raw_parts_mut(ptr, tbs.len())
-        },
-        TbsType::FmcaliasTbs => unsafe {
-            let ptr = &mut FMCALIAS_TBS_ORG as *mut u8;
-            core::slice::from_raw_parts_mut(ptr, tbs.len())
-        },
+        TbsType::LdevidTbs => {
+            env.fht_data_store.ldevid_tbs_size = tbs.len() as u16;
+            unsafe {
+                let ptr = &mut LDEVID_TBS_ORG as *mut u8;
+                core::slice::from_raw_parts_mut(ptr, tbs.len())
+            }
+        }
+        TbsType::FmcaliasTbs => {
+            env.fht_data_store.fmcalias_tbs_size = tbs.len() as u16;
+            unsafe {
+                let ptr = &mut FMCALIAS_TBS_ORG as *mut u8;
+                core::slice::from_raw_parts_mut(ptr, tbs.len())
+            }
+        }
     };
 
     dst[..tbs.len()].copy_from_slice(tbs);
```

Comment:
Shall we error out if tbs.len() is greater than the TBS region size?

Reply:
I would recommend doing something like this (can be done in a follow up PR if we want)

```rust
extern "C" {
    static mut LDEVID_TBS_ORG: [u8; LDEVID_TBS_SIZE];
    static mut FMCALIAS_TBS_ORG: [u8; FMCALIAS_TBS_SIZE];
}

...

let dst = match tbs_type {
    TbsType::LdevidTbs => {
            env.fht_data_store.ldevid_tbs_size = tbs.len() as u16;
            unsafe { &LDEVID_TBS_ORG[..] }
        }
        TbsType::FmcaliasTbs => {
            env.fht_data_store.fmcalias_tbs_size = tbs.len() as u16;
            unsafe { &FMCALIAS_TBS_ORG[..] }
        }
};
```

This makes the bounds checking easier and ensures the compiler helps you out (by inserting a panic which our test should check).

Reply:
Will fix it as part of https://github.com/chipsalliance/caliptra-sw/issues/354